### PR TITLE
MM-316 Allows users to drag and drop files from their computer to the center pane or RHS to upload them

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Also fixes an issue with uploading files to the RHS after switching channels along with some minor cosmetic reformatting of file_upload.jsx.

I added a small jquery 'plugin' to help handle the actual drag and drop detection rather than use the not so great default html5 implementation (it's really not good).

Finally, thanks to @asaadmahmoodspin for the UI changes!

### CODE REVIEW

File: web/sass-files/sass/partials/_post.scss [hunk 1]
```diff
@@ -106,6 +106,39 @@ body.ios {
...
+	}
+
+	.fa {
+		display: block;
+		font-size: 2em;
+		margin: 0 0 0.3em;
+	}
+
+	.invisible {
+		visibility: hidden;
```
Comment (ralder) [inline web/sass-files/sass/partials/_post.scss]: this rule already exists in bootstrap

File: web/sass-files/sass/partials/_post.scss [hunk 2]
```diff
@@ -106,6 +106,39 @@ body.ios {
...
+	}
+	.visible {
+		visibility: visible;
```
Comment (ralder) [inline web/sass-files/sass/partials/_post.scss]: seems this rule not need because may use just toggle classname `.invisible`

Comment (rgarmsen2295) [inline web/sass-files/sass/partials/_post.scss]: Thanks for the comments @ralder I think you're right for both of these let me take a look and see about changing these.

**Comment by rgarmsen2295** [comment]
Okay thanks again @ralder, implemented your suggestions and fixed an issue that popped up with @asaadmahmoodspin's help.  @jwilander please review at your leisure, thanks!

**Comment by jwilander** [comment]
Code looks good to me. What browsers was this tested on?

**Comment by rgarmsen2295** [comment]
Chrome on osx, safari, and IE10 through a mac vm running win7; part of the reason I went with the jquery version of the plugin was the better cross-browser support.

**Comment by jwilander** [comment]
Perfect. Just waiting for a seconder

**Comment by crspeller** [comment]
Looks good!

**Pull request merged**